cmd/hvclient/request_builders.go

```diff
@@ -138,6 +140,15 @@ func buildRequest(reqinfo *requestValues) (*hvclient.Request, error) {
 		return nil, err
 	}
 
+	// Only add the signature hash algorithm if specified, otherwise we don't
+	// want to bother sending out an object.
+	if reqinfo.sigAlg != "" || reqinfo.sigHash != "" {
```

Comment:
Do the omitempty's around the Signature, Algorithm and HashAlgorithm not make this redundant?